### DISCUSSION TITLE
auto generated router is global ? or how import different router folder  ?

### DISCUSSION THREAD

**Discussion by wildwind123** [posva/unplugin-vue-router#422] [Questions and Help] [answered]
in multi page app
https://vitejs.dev/guide/build#multi-page-app
in entrypoint js can config like this in non file based routing

in this case i can setting separate router logic in every entrypoint

```ts
import { createMemoryHistory, createRouter } from 'vue-router'

import HomeView from './HomeView.vue'
import AboutView from './AboutView.vue'

const routes = [
  { path: '/', component: HomeView },
  { path: '/about', component: AboutView },
]

const router = createRouter({
  history: createMemoryHistory(),
  routes,
})
```

in file based get from global autogenerated file
like this
```ts
import { routes } from 'vue-router/auto-routes'
```

and i cant setting separate router logic like in non file based routing.

**Comment by posva** [comment, marked as answer]
I imagine you could setup typescript per app in order to scope the types to each app.

**Reply by wildwind123** [reply]
Thanks.

I have idea. This approach, that would be enough for me. Just like idea. 
Config like this, every item of routesFolder will have name.
```ts
VueRouter({
  routesFolder: [
    {
      name: admin,
      src: 'src/admin/routes',
      path: 'admin/',
    },
    {
      name: docs,
      src: 'src/docs',
      path: 'docs/:lang/',
    },
    {
      name: promos,
      src: 'src/promos',
      path: 'promos-',
    },
  ],
})
```

and on import will like this, and in every entrypoint will import only necessary routes

```ts
// import all
import { admin, docs, promos } from 'vue-router/auto-routes'
```
